File: frontend/lib/models/restaurant.dart
class MatchedRestaurant {
  final String id;
  final String name;
  final double? avgRating;
  final String? totalRatings;
  final int? deliveryTimeMinutes;
  final String? deliveryTimeRange;
  final String? costForTwo;
  final String? availabilityStatus;
  final String? nextOpenTime;
  final String? imageUrl;
  final List<String> cuisines;
  final String? areaName;
  final double? distanceKm;
  final List<String> matchedPreferences;
  final String appDeepLink;
  final String webUrl;
  final String? offer;

  const MatchedRestaurant({
    required this.id,
    required this.name,
    this.avgRating,
    this.totalRatings,
    this.deliveryTimeMinutes,
    this.deliveryTimeRange,
    this.costForTwo,
    this.availabilityStatus,
    this.nextOpenTime,
    this.imageUrl,
    required this.cuisines,
    this.areaName,
    this.distanceKm,
    required this.matchedPreferences,
    required this.appDeepLink,
    required this.webUrl,
    this.offer,
  });

  factory MatchedRestaurant.fromJson(Map<String, dynamic> json) =>
      MatchedRestaurant(
        id: json['id'] as String,
        name: json['name'] as String,
        avgRating: (json['avgRating'] as num?)?.toDouble(),
        totalRatings: json['totalRatings'] as String?,
        deliveryTimeMinutes: json['deliveryTimeMinutes'] as int?,
        deliveryTimeRange: json['deliveryTimeRange'] as String?,
        costForTwo: json['costForTwo'] as String?,
        availabilityStatus: json['availabilityStatus'] as String?,
        nextOpenTime: json['nextOpenTime'] as String?,
        imageUrl: json['imageUrl'] as String?,
        cuisines: (json['cuisines'] as List<dynamic>?)
                ?.map((e) => e as String)
                .toList() ??
            [],
        areaName: json['areaName'] as String?,
        distanceKm: (json['distanceKm'] as num?)?.toDouble(),
        matchedPreferences: (json['matchedPreferences'] as List<dynamic>)
            .map((e) => e as String)
            .toList(),
        appDeepLink: json['appDeepLink'] as String,
        webUrl: json['webUrl'] as String,
        offer: json['offer'] as String?,
      );
}

class MatchResponse {
  final List<String> preferences;
  final int matchCount;
  final List<MatchedRestaurant> restaurants;

  const MatchResponse({
    required this.preferences,
    required this.matchCount,
    required this.restaurants,
  });

  factory MatchResponse.fromJson(Map<String, dynamic> json) => MatchResponse(
        preferences:
            (json['preferences'] as List<dynamic>).map((e) => e as String).toList(),
        matchCount: json['matchCount'] as int,
        restaurants: (json['restaurants'] as List<dynamic>)
            .map((e) => MatchedRestaurant.fromJson(e as Map<String, dynamic>))
            .toList(),
      );
}

File: order-together-backend/src/main/java/com/ordertogether/model/Restaurant.java
package com.ordertogether.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

/**
 * A restaurant as returned by the Swiggy MCP {@code search_restaurants} tool.
 *
 * <p>{@code costForTwo} is kept as a {@link String} because Swiggy returns it formatted
 * (e.g. {@code "₹300 for two"}). Unknown fields are ignored so the model stays
 * forward-compatible with the MCP payload.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record Restaurant(
		String id,
		String name,
		Double avgRating,
		String totalRatings,
		Integer deliveryTimeMinutes,
		String deliveryTimeRange,
		String costForTwo,
		String availabilityStatus,
		String nextOpenTime,
		String imageUrl,
		List<String> cuisines,
		String areaName,
		Double distanceKm) {

	/** True unless the restaurant explicitly reports any CLOSED availability status. */
	public boolean isOpen() {
		return availabilityStatus == null
				|| !availabilityStatus.trim().toUpperCase().contains("CLOSED");
	}
}
